Modules/_zstd/compressor.c

```diff
@@ -412,8 +422,8 @@ ZstdCompressor_dealloc(PyObject *ob)
 }
 
 static PyObject *
-compress_impl(ZstdCompressor *self, Py_buffer *data,
-              ZSTD_EndDirective end_directive)
+compress_lock_held(ZstdCompressor *self, Py_buffer *data,
+                   ZSTD_EndDirective end_directive)
```

Comment:
Add `assert(PyMutex_IsLocked(&self->lock));` to lock held functions to ensure the lock is held.